### PR TITLE
chore(deps): update OpenShell to 0.0.106

### PR DESCRIPTION
**Pull request by prekshivyas**
## Summary

Updates OpenShell from 0.0.101 to 0.0.106 using the sealed NemoPin migration evidence. Release-trust prerequisite #9224 is merged on `main`, and the E2E-selector work from #9378 is included in this branch. The PR is reconciled with current `main` at latest PR commit `e3a53c96f`.

## Related Issue

- #6256: owner decision
- #3136: open upstream work
- #6871: open upstream work
- #7367: open upstream work
- #7937: open upstream work
- #7957: open upstream work
- #8769: open upstream work
- #8887: open upstream work
- #8893: open upstream work

## Changes

- Applies only paths authorized by `sha256:65355e5c4180f3716fd8e0d0431d432a876460a7005db6099fa18a8599c3037e`.
- Migrates the exact base `44c2636d85f788f81767fbb451717566e0a8d475` across 5 adjacent release ranges.
- Reconciles the migration with current `main` at base commit `40dc27283`, includes the reviewed E2E-selector work from #9378, adopts the upstream messaging-plan fixture fix from #9517, and hardens runtime-identity qualification for cold 0.0.106 onboarding while recording the attached-provider post-state.
- Changed paths: `.github/workflows/e2e.yaml`, `.github/workflows/podman-cpu-proof.yaml`, `agents/hermes/Dockerfile`, `agents/hermes/mcp-config-transaction.py`, `agents/hermes/runtime-config-guard.py`, `agents/hermes/start.sh`, `docs/deployment/set-up-mcp-bridge.mdx`, `docs/manage-sandboxes/add-mcp-server.mdx`, `docs/manage-sandboxes/update-sandboxes.mdx`, `docs/reference/commands.mdx`, `docs/reference/configure-runtime-identity.mdx`, `docs/reference/troubleshooting.mdx`, `docs/security/best-practices.mdx`, `docs/security/gateway-authentication-controls.mdx`, `internal/security-reviews/openshell-0.0.72-compatibility-review.mdx`, `nemoclaw-blueprint/blueprint.yaml`, `nemoclaw/src/shared/openshell-policy-boundary.cts`, `scripts/brev-launchable-ci-cpu.sh`, `scripts/checks/dependency-pins.mts`, `scripts/checks/managed-image-protected-runtime-contract.ts`, `scripts/install-openshell.sh`, `scripts/install.sh`, `scripts/update-hermes-agent.sh`, `src/lib/actions/sandbox/mcp-bridge-input-validation.test.ts`, `src/lib/actions/sandbox/mcp-bridge-url-validation.ts`, `src/lib/actions/sandbox/mcp-bridge-validation.ts`, `src/lib/actions/sandbox/openshell-child-visible-credentials.v0.0.106.json`, `src/lib/onboard/docker-driver-gateway-config-toml.test.ts`, `src/lib/onboard/docker-driver-gateway-runtime.test.ts`, `src/lib/onboard/docker-driver-gateway-runtime.ts`, `src/lib/onboard/experimental/portable-demo-lifecycle.ts`, `src/lib/onboard/forward-start.ts`, `src/lib/onboard/managed-bootstrap/podman-held-workload.test.ts`, `src/lib/onboard/managed-bootstrap/podman-held-workload.ts`, `src/lib/onboard/openshell-feature-gate.test.ts`, `src/lib/onboard/openshell-feature-gate.ts`, `src/lib/onboard/openshell-install.test.ts`, `src/lib/onboard/openshell-install.ts`, `src/lib/onboard/openshell-version.ts`, `src/lib/onboard/runtime-provider/podman-lifecycle.ts`, `src/lib/policy/index.ts`, `test/brev-launchable-ci-cpu-checksum.test.ts`, `test/deepagents-mcp-legacy-lifecycle.test.ts`, `test/dependency-pins-check.test.ts`, `test/e2e/fixtures/openshell-v0106-qualification.ts`, `test/e2e/fixtures/security-posture.ts`, `test/e2e/live/mcp-bridge-sandbox.ts`, `test/e2e/live/network-policy.test.ts`, `test/e2e/live/openclaw-plugin-runtime-exdev.test.ts`, `test/e2e/live/openshell-gateway-auth-source-contract-helpers.ts`, `test/e2e/live/openshell-gateway-auth-source-contract.test.ts`, `test/e2e/live/openshell-gateway-upgrade.test.ts`, `test/e2e/live/openshell-v0106-tls-server-name-source.ts`, `test/e2e/live/podman-cpu-lifecycle-helpers.ts`, `test/e2e/live/podman-cpu-lifecycle.test.ts`, `test/e2e/support/mcp-bridge-runtime-compatibility.test.ts`, `test/e2e/support/mcp-bridge-sandbox.test.ts`, `test/e2e/support/mcp-workflow-boundary.test.ts`, `test/e2e/support/openshell-gateway-auth-contract-workflow-boundary.test.ts`, `test/e2e/support/openshell-v0106-tls-server-name-source.test.ts`, `test/e2e/support/workflow-plan.test.ts`, `test/exit-code-user-error-surfaces.test.ts`, `test/fixtures/openshell-v0.0.106`, `test/gateway-state-reconcile-2276.test.ts`, `test/hermes-doctor-config-hash.test.ts`, `test/hermes-mcp-config-transaction.test.ts`, `test/hermes-mcp-credential-boundary-manifest.test.ts`, `test/install-openshell-version-check.test.ts`, `test/install-openshell-version-pin.test.ts`, `test/installer-hash-check.test.ts`, `test/installer-sandbox-build-trust.test.ts`, `test/installer-supervisor-manifest-trust.test.ts`, `test/mcp-add-crash-consistency.test.ts`, `test/mcp-destroy-lifecycle.test.ts`, `test/mcp-policy-key-ownership.test.ts`, `test/mcp-restart-policy-order.test.ts`, `test/onboard-gateway-port-conflict-fast-fail.test.ts`, `test/openshell-0.0.85-migration-review.test.ts`, `test/openshell-channel-workflow.test.ts`, `test/pr-risk-plan.test.ts`, `test/rebuild-credential-preflight.test.ts`, `test/runner.test.ts`, `test/sandbox-provisioning.test.ts`, `test/sandbox-rlimit-hooks.test.ts`, `test/update-hermes-agent-script.test.ts`, `tools/e2e/mcp-workflow-boundary.mts`, `tools/e2e/openshell-gateway-auth-contract-workflow-boundary.mts`, `tools/e2e/workflow-boundary.mts`

### Release ranges

| Range | Commits | State | Concerns |
|---|---|---|---|
| 0.0.101 → v0.0.102 | `8ddd98c3dff6` → `f48b05e31228` | published | 3 |
| v0.0.102 → v0.0.103 | `f48b05e31228` → `c825b1f8efac` | published | 2 |
| v0.0.103 → v0.0.104 | `c825b1f8efac` → `dd2b4e3bc068` | published | 1 |
| v0.0.104 → v0.0.105 | `dd2b4e3bc068` → `0f8fad23c471` | published | 3 |
| v0.0.105 → v0.0.106 | `0f8fad23c471` → `c4b500a7de64` | published | 5 |

### Concern dispositions

| Concern | Surface | Planned disposition | Failure prevented | Remaining gate |
|---|---|---|---|---|
| `openshell-0.0.101..v0.0.102-network-1` | network | test | v0.0.102 reports network: ### Quick install * fix(e2e): separate Podman Machine loopback listeners by @matthewgrossman in https://github.com/NVIDIA/OpenShell/pull/2622 * fix(pol... | none |
| `openshell-0.0.101..v0.0.102-runtime-topology-2` | runtime topology | test | v0.0.102 reports runtime topology: ### Quick install * fix(e2e): separate Podman Machine loopback listeners by @matthewgrossman in https://github.com/NVIDIA/OpenShell/pull/2622 ... | runtime-proof |
| `openshell-0.0.101..v0.0.102-security-identity-3` | security identity | guard | v0.0.102 reports security identity: ### Quick install * fix(e2e): separate Podman Machine loopback listeners by @matthewgrossman in https://github.com/NVIDIA/OpenShell/pull/2622... | none |
| `openshell-v0.0.102..v0.0.103-network-1` | network | test | v0.0.103 reports network: ### Quick install * fix(sandbox): acknowledge unchanged policy revisions by @NaveCohenMonday in https://github.com/NVIDIA/OpenShell/pull/2557 * fix(gat... | none |
| `openshell-v0.0.102..v0.0.103-runtime-topology-2` | runtime topology | test | v0.0.103 reports runtime topology: ### Quick install * fix(sandbox): acknowledge unchanged policy revisions by @NaveCohenMonday in https://github.com/NVIDIA/OpenShell/pull/2557 ... | runtime-proof |
| `openshell-v0.0.103..v0.0.104-runtime-topology-1` | runtime topology | test | v0.0.104 reports runtime topology: ### Quick install * feat(build): add glibc-static supervisor libc variant by @EmilienM in https://github.com/NVIDIA/OpenShell/pull/2682 * fix(... | runtime-proof |
| `openshell-v0.0.104..v0.0.105-compatibility-change-1` | compatibility change | test | v0.0.105 reports compatibility change: ### Quick install * fix(gator): separate review budget from approval gate by @johntmyers in https://github.com/NVIDIA/OpenShell/pull/2704 ... | none |
| `openshell-v0.0.104..v0.0.105-network-2` | network | test | v0.0.105 reports network: ### Quick install * fix(gator): separate review budget from approval gate by @johntmyers in https://github.com/NVIDIA/OpenShell/pull/2704 * perf(superv... | none |
| `openshell-v0.0.104..v0.0.105-runtime-topology-3` | runtime topology | test | v0.0.105 reports runtime topology: ### Quick install * fix(gator): separate review budget from approval gate by @johntmyers in https://github.com/NVIDIA/OpenShell/pull/2704 * pe... | runtime-proof |
| `openshell-v0.0.105..v0.0.106-lifecycle-state-1` | lifecycle state | test | v0.0.106 reports lifecycle state: ### Quick install * ci(cargo-deny): add dependency audit with cargo-deny by @Ygnas in https://github.com/NVIDIA/OpenShell/pull/2677 * feat(sdk/... | runtime-proof |
| `openshell-v0.0.105..v0.0.106-code-impact-configuration-1` | mapped configuration | test | The exact-ref diff reports configuration changes in crates/openshell-cli/src/commands/common.rs, crates/openshell-cli/src/main.rs, crates/openshell-cli/src/run.rs. Mapped NemoCl... | runtime-proof |
| `openshell-v0.0.105..v0.0.106-code-impact-contract-or-schema-2` | mapped contract or schema | test | The exact-ref diff reports contract or schema changes in crates/openshell-core/src/middleware.rs, crates/openshell-core/src/provider_credentials.rs, crates/openshell-driver-kube... | runtime-proof |
| `openshell-v0.0.105..v0.0.106-code-impact-security-3` | mapped security | test | The exact-ref diff reports security changes in crates/openshell-bootstrap/src/build_windows.rs, crates/openshell-cli/src/commands/common.rs, crates/openshell-cli/src/main.rs. Ma... | runtime-proof |
| `openshell-v0.0.105..v0.0.106-code-impact-test-4` | mapped test | test | The exact-ref diff reports test changes in crates/openshell-cli/tests/ensure_providers_integration.rs, crates/openshell-cli/tests/mtls_integration.rs, crates/openshell-cli/tests... | runtime-proof |

### Immutable artifacts

| Artifact | SHA-256 |
|---|---|
| openshell-aarch64-apple-darwin.tar.gz | `969493205e3d3462…` |
| openshell-aarch64-unknown-linux-musl.tar.gz | `ce981904ae8febd9…` |
| openshell-checksums-sha256.txt | `7421aaf9d5550dc1…` |
| openshell-gateway-aarch64-apple-darwin.tar.gz | `de8f90db9dd0d3b4…` |
| openshell-gateway-aarch64-unknown-linux-gnu.tar.gz | `22b7781249e34870…` |
| openshell-gateway-checksums-sha256.txt | `26e4345449e02475…` |
| openshell-gateway-x86_64-unknown-linux-gnu.tar.gz | `b7760cb752a4363c…` |
| openshell-gateway-x86_64-unknown-linux-gnu.tar.gz::executable:openshell-gateway | `e6cde8a54568aa19…` |
| openshell-sandbox-aarch64-unknown-linux-gnu.tar.gz | `5e5d758d53c6abc6…` |
| openshell-sandbox-aarch64-unknown-linux-gnu.tar.gz::executable:openshell-sandbox | `0031c6b257a23ecc…` |
| openshell-sandbox-checksums-sha256.txt | `88bc98ffdc915fb7…` |
| openshell-sandbox-x86_64-unknown-linux-gnu.tar.gz | `559b8aaad3a8eeab…` |
| openshell-sandbox-x86_64-unknown-linux-gnu.tar.gz::executable:openshell-sandbox | `019301ec8618abbe…` |
| openshell-x86_64-unknown-linux-musl.tar.gz | `d1a885a91b3e5aaa…` |
| openshell-x86_64-unknown-linux-musl.tar.gz::executable:openshell | `98ecf95113fea999…` |
| openshell.rb | `f0f86519e227b3b3…` |
| openshell-source:crates/openshell-core/src/google_cloud.rs | `2583a04a0557f069…` |
| openshell-source:crates/openshell-core/src/provider_credentials.rs | `7e8e05efcb725807…` |
| openshell-source:crates/openshell-core/src/secrets.rs | `b122b4a5af5a5823…` |
| ghcr.io/nvidia/openshell/supervisor:index | `722f44669722961b…` |

### Validation receipt

| Gate | Current result |
|---|---|
| targeted | Passed: 609 assertions with 1 intentional skip across the migration/runtime suites, 290/290 E2E registry/workflow overlap assertions, and 73/73 onboarding-preflight overlap assertions on `cc9167892`. After the final main reconciliation, 32/32 conflict-sensitive gateway/TLS assertions and `npm run validate:pr` passed on latest PR commit `d1990537d`. |
| brev-integration-fresh-onboarding | Passed on Brev before reconciliation; latest PR commit GitHub CI passed |
| brev-integration-existing-upgrade | Passed on Brev before reconciliation; latest PR commit GitHub CI passed |
| brev-integration-recovery-rollback | Passed on Brev before reconciliation; latest PR commit GitHub CI passed |
| full-e2e | Exact inference run [32199961381](https://github.com/NVIDIA/NemoClaw/actions/runs/32199961381) on PR commit `cc9167892` with trusted base/workflow `a9fc8045d` passed 5/7 cases. TC-INF-12 and TC-INF-13 both completed cold onboarding, plan/apply, provider attachment, token refresh, and post-attach inference; OpenShell 0.0.106 then failed to project `E2E_ACCESS_TOKEN` / `ENTRA_ACCESS_TOKEN` into 19 fresh sandbox execs over 35 seconds. Protected managed-image runtime was not launched because inference is its prerequisite. This is a repeated upstream runtime-credential projection blocker, not a waived gate. |

## Type of Change

- [ ] Code change (feature, bug fix, or refactor)
- [x] Code change with doc updates
- [ ] Doc only (prose changes, no code sample modifications)
- [ ] Doc only (includes code sample changes)

## Quality Gates

- [x] Tests added or updated for changed behavior
- [ ] Existing tests cover changed behavior — justification:
- [ ] Tests not applicable — justification:
- [x] Docs updated for user-facing behavior changes
- [ ] Docs not applicable — justification:
- [x] Sensitive paths changed (security, policy, credentials, preflight, onboarding, inference, runner, sandbox, or messaging)
- [x] Sensitive-path review completed or maintainer-approved waiver recorded — reviewer/approval link/justification: Independent exact-commit review passed for the OpenShell 0.0.106 migration and stacked E2E selector.
- [ ] Non-success, skipped, or missing CI check accepted by maintainer — check name, approval link, and follow-up issue:

## Documentation Writer Review

- [x] Documentation writer subagent reviewed the completed changes
- Result: `docs-updated`
- Evidence: `docs/deployment/set-up-mcp-bridge.mdx`, `docs/manage-sandboxes/add-mcp-server.mdx`, `docs/manage-sandboxes/manage-mcp-servers.mdx`, `docs/reference/commands.mdx`, and `docs/reference/troubleshoot-mcp-servers.mdx`
- Agent: Codex Desktop

## DGX Station Hardware Evidence

- [ ] Tested on DGX Station
- Tested commit:
- Station profile/scenario:
- Result:
- Supporting evidence:

## Verification

- [x] PR description includes a `Signed-off-by:` line and every commit appears as `Verified` in GitHub
- [x] Normal `pre-commit`, `commit-msg`, and `pre-push` hooks passed, or `npm run validate:pr` passed after refreshing `origin/main` when hooks were skipped or unavailable
- [x] Targeted behavior tests pass for the current change set, or tests are marked not applicable above — 609 assertions passed with 1 intentional skip across the migration/runtime suites, followed by 290/290 E2E registry/workflow overlap assertions and 73/73 onboarding-preflight overlap assertions on `cc9167892`; 32/32 conflict-sensitive gateway/TLS assertions and `npm run validate:pr` then passed after the final main reconciliation on latest PR commit `d1990537d`.
- [x] Applicable broad gate passed — GitHub reports no required checks for this branch. Optional checks do not gate review unless they expose a PR defect.
- [x] Quality Gates section completed with required justifications or waivers
- [x] No secrets, API keys, or credentials committed
- [ ] `npm run docs` builds without warnings (doc changes only)
- [x] Doc pages follow the [style guide](https://github.com/NVIDIA/NemoClaw/blob/main/docs/CONTRIBUTING.md) (doc changes only)
- [ ] New doc pages include SPDX header and frontmatter (new pages only)

---

Signed-off-by: Prekshi Vyas <prekshiv@nvidia.com>

## NemoPatch current-head status

- Status: **blocked**
- Latest PR commit: `d1990537d9b9c8579c3197cba33faddaeb9aa101`
- Checked: 2026-08-19T02:46:48Z

- Reconciled with `main` at base commit `ee6762b99`; 32/32 conflict-sensitive gateway/TLS assertions, local PR validation, and all 45 applicable current-commit GitHub checks pass. The two advisor services failed analysis without findings. Exact inference on `cc9167892` with trusted base/workflow `a9fc8045d` passed 5/7 and repeatedly isolated an OpenShell 0.0.106 runtime-credential projection failure after successful provider attachment and refresh. Protected managed-image E2E remains gated; maintainer/upstream resolution is required.

## Summary by CodeRabbit

- **New Features**
  - Added support for OpenShell `0.0.106`, including updated CLI, gateway, sandbox, supervisor, and credential-boundary assets.
  - Added stronger installation integrity checks for reviewed binaries and release artifacts.
  - Added TLS server-name verification across Docker, Podman, and VM environments.

- **Bug Fixes**
  - Improved MCP bridge destination validation before connections are established.
  - Added supervisor TLS-name sanitization protections.

- **Documentation**
  - Updated setup, compatibility, security, installation, and troubleshooting guidance for OpenShell `0.0.106`.
  - Added guidance for resolving Docker credential-store failures in headless environments.

### CODE REVIEW

**Review by prekshivyas — COMMENTED** [review]
Reviewed exact head 26248e14db0d0c17f9c0a268b6ae9e0509aee14f after merging current main. The conflict resolution retains the main 0.0.103 trust records and the candidate 0.0.105 manifest records, and focused dependency tests plus shell syntax checks pass locally. This draft cannot become green or approvable yet: the base-trusted installer parser on main has no reviewed 0.0.105 standalone-sandbox or supervisor identities, so candidate code cannot self-authorize them; check-hash fails closed as designed. The protected Podman proof also still dispatches OpenShell 0.0.101 while the candidate installer intentionally accepts only 0.0.105. Land the prerequisite trust identities on main and update the protected workflow pin through the authorized dependency process, then refresh this branch. GitHub also prohibits me from approving my own PR.

**Review by jyaunches — DISMISSED** [review]
## LOC Reduction / Codebase Simplicity Review

### Why this blocks

- `test/e2e/live/openshell-v0106-tls-server-name-source.ts` defines parallel source and regression schemas.
- It duplicates size, hash, ordering assertions and fetch loops.
- Podman and VM source and regression entries use identical `{path, blobSha}`, so reviewed blobs are fetched and hashed twice.

### Refactor direction

- Represent each unique `{path, blobSha}` once with one or more ordered-token checks.
- Use one bounded fetch and hash loop, then apply every check attached to that source.
- Preserve separate result categories only in the output projection.

### Expected result

- Reduce six source requests to four.
- Remove duplicate interfaces, assertion functions, and fetch loops while retaining the same trusted evidence.

**Review by jyaunches — COMMENTED** [review]
## LOC Reduction / Codebase Simplicity Review

### Resolved

Resolved at current exact head `cf5b1b9bc057684673c460f24c5eb1798016f6f5`.

The requested consolidation arrived in `6d22b012e0e84860b59c73ef99ba677a104c4275`, and both affected files remain byte-identical after the latest main refresh.

### Simplification result

- Each unique `{path, blobSha}` is represented once.
- Podman and VM attach both driver and regression ordered-token checks to their single reviewed source.
- One assertion function owns byte-limit, blob-hash, and token-order validation.
- One bounded loop fetches and hashes the four unique sources.
- Driver and regression categories are preserved only in the result projection.
- The focused regression test proves exactly four source requests.

The source verifier drops from 214 to 196 lines, and its focused test drops from 131 to 130 lines while adding the explicit four-fetch assertion.

### Review outcome

The duplicate schemas, validators, and fetch loops identified by the active request are gone. I found no replacement LOC-reduction or codebase-simplicity blocker.

This closes only the LOC/codebase-simplicity review. It is not an approval or a judgment on other review dimensions.

**Review by jyaunches — DISMISSED** [review]
## LOC Reduction / Codebase Simplicity Review

### Why this blocks

- The commit under review sets both supported OpenShell pins to `0.0.106`, but retains `expectedOpenShellE2eVersion()` and 30 test lines for the completed `0.0.101` qualification transition. That exception has no current repository state that can select it.
- `tools/e2e/install-openshell-v00106-qualification.sh` adds 84 lines of a second OpenShell installer. It repeats the release version, three archive hashes, three binary hashes, download logic, archive validation, installation, and version validation already owned by `scripts/install-openshell.sh`.
- The qualification installer states that it serves jobs before NemoClaw changes its supported `0.0.101` pin. This PR changes that pin to `0.0.106`.
- Workflow validators and changed-file routing add more version-specific configuration and tests around this duplicate installer.

### Refactor direction

- Route the affected E2E jobs through `scripts/install-openshell.sh`. The canonical installer now supports `0.0.106` and installs the same CLI, gateway, and sandbox artifacts.
- Keep required environment removal and exact-version settings at each workflow call site.
- Remove the version-specific qualification installer and its routing and installer-shape assertions.
- Compare the workflow version directly with `pins.maxVersion`. Remove `expectedOpenShellE2eVersion()` and its `0.0.101` transition tests.
- Retain only release or source evidence that has an independent post-migration consumer.

### Expected result

- One installer owns downloads, archive validation, artifact hashes, installation, and version checks.
- One blueprint-derived version authority drives the workflow pin.
- Remove at least 122 added lines from the duplicate installer and completed transition exception, plus the associated routing coverage.
- Preserve the required OpenShell `0.0.106` E2E behavior with fewer release-specific maintenance paths.

**Comment by prekshivyas** [comment]
Addressed the active LOC/codebase-simplicity review at latest PR commit `07ed4657194dabc774130b8368ad11b0e79a0af6`:

- removed the 84-line version-specific qualification installer and routed all affected jobs through `scripts/install-openshell.sh`
- preserved credential removal, forced stable installation, exact `0.0.106` workflow settings, and independently consumed release/source provenance checks at the call sites
- removed `expectedOpenShellE2eVersion()`, its completed `0.0.101` transition tests, and the deleted-installer routing/shape assertions
- restored the historical `0.0.72` security review to release-scoped evidence
- reduced the reviewer update by 238 net lines (61 additions, 299 deletions)

Validation on this exact commit: 313 focused assertions passed with 1 intentional skip; dependency pin, formatting, E2E semantic-plan, source-shape, growth-guardrail, repository, lint, secret-scan, commit, and push hooks passed. GitHub reports the commit signature verified. Fresh CI and exact-commit E2E are pending.

@jyaunches, please re-review the resolved simplicity blocker when convenient.

File: internal/security-reviews/openshell-0.0.72-compatibility-review.mdx [hunk 1]
```diff
@@ -33,11 +33,13 @@ The generated gateway authentication contract remains unchanged from the OpenShe
...
+For the current OpenShell 0.0.106 pin, the same scenario verifies the exact release-source blobs for the Docker, Podman, and VM drivers and requires each driver to remove `OPENSHELL_GATEWAY_TLS_SERVER_NAME` after merging user environment.
+It also verifies the exact upstream regression blobs that inject a hostile server-name value through each driver's sandbox specification and require the value to be absent from the resulting supervisor or guest environment.
 The inherited contract also continues to reject `NEMOCLAW_GATEWAY_BIND_ADDRESS=0.0.0.0`.
 User principals remain blocked from sandbox-only methods.
 
 The compatibility container remains an explicit trusted-host fallback behind `NEMOCLAW_OPENSHELL_GATEWAY_CONTAINER_PATCH=1`.
-It uses host networking and read-only Docker socket access, so hosts with glibc 2.39 or newer remain preferred for the current OpenShell 0.0.101 pin.
+It uses host networking and read-only Docker socket access, so hosts with glibc 2.39 or newer remain preferred for the current OpenShell 0.0.106 pin.
```
[the remark this answers is not in the document]
Comment (prekshivyas) [reply internal/security-reviews/openshell-0.0.72-compatibility-review.mdx]: Resolved in `07ed46571` by removing the two 0.0.106 source/regression evidence paragraphs from this historical review and changing the host-preference sentence back to the reviewed OpenShell 0.0.72 pin. The page is now consistently scoped to 0.0.72; current 0.0.106 evidence remains in its independently consumed migration fixtures and workflow provenance checks.

File: test/e2e/support/mcp-workflow-boundary.test.ts [hunk 2]
```diff
@@ -12,6 +12,40 @@ import { validateMcpOpenShellWorkflowBoundary } from "../../../tools/e2e/mcp-wor
...
+    const workflowPath = path.join(directory, "e2e.yaml");
+    try {
+      const workflow = YAML.parse(fs.readFileSync(".github/workflows/e2e.yaml", "utf8")) as {
+        jobs: Record<string, { steps: Array<{ name?: string; run?: string }> }>;
+      };
+      const install = workflow.jobs[jobName].steps.find(
+        (step) => step.name === "Install OpenShell CLI",
+      );
+      requireFixture(install?.run, `${jobName} qualification installer fixture is missing`);
+      install.run += "bash scripts/install-openshell.sh\n";
```
[the remark this answers is not in the document]
Comment (prekshivyas) [reply test/e2e/support/mcp-workflow-boundary.test.ts]: Resolved in `07ed46571` by removing this completed-transition test together with the duplicate version-specific installer and its installer-shape assertions, as requested by the active simplicity review. The affected jobs now invoke only the canonical installer, and focused workflow-boundary tests pass.

**Review by jyaunches — COMMENTED** [review]
## LOC Reduction / Codebase Simplicity Review

### Resolved

Resolved at commit under review `07ed4657194dabc774130b8368ad11b0e79a0af6`.

### Simplification result

- The 84-line version-specific qualification installer is deleted.
- The three affected E2E jobs now use `scripts/install-openshell.sh`.
- `expectedOpenShellE2eVersion()` and its 30-line completed-transition test matrix are deleted.
- Installer-specific routing and duplicate installer-shape coverage are removed.
- The workflow validators now require the repository OpenShell installer.

The refactor changes 61 lines and removes 299 lines, for a net reduction of 238 lines.

### Review outcome

The duplicate installer and completed `0.0.101` transition authority identified by the active request are gone. I found no replacement LOC-reduction or codebase-simplicity blocker.

This closes only the LOC/codebase-simplicity review. It is not an approval or a judgment on other review dimensions.

**Review by jyaunches — DISMISSED** [review]
## LOC Reduction / Codebase Simplicity Review

### Why this blocks

The latest PR commit includes the dashboard-port reservation fix from a separate issue and open pull request.

Commit `2c062d115b4f64fc73d6685b827479c41ec36a84` changes `src/lib/onboard/dashboard-port.ts` and adds the extracted-callback loopback test. Those are the same source correction and regression contract owned by PR #9569 for accepted issue #9568. They are unrelated to this OpenShell `0.0.106` migration and add a second open implementation of the same fix.

Keeping the dashboard change here makes the dependency migration and the focused bug fix conflict and forces one behavior to be reviewed, merged, or reverted through two pull requests.

### Refactor direction

- Remove the dashboard-port source and test changes from this branch.
- Let PR #9569 remain the single owner for issue #9568.
- Keep this pull request limited to the OpenShell migration and its directly required compatibility evidence.

### Expected result

One pull request owns the dashboard reservation correction and one pull request owns the OpenShell migration. The current branch loses the unrelated duplicate test and source delta without changing its dependency-migration behavior.

**Comment by prekshivyas** [comment]
Addressed the scope request in signed commit `5e5486fadddf819824bdce5d6c8f47383a1e1c79`: removed the dashboard-port source and regression-test delta from this branch. Both files now match `main`; PR #9569 remains the single owner of issue #9568.

**Review by jyaunches — COMMENTED** [review]
## LOC Reduction / Codebase Simplicity Review

### Resolved

The duplicate dashboard-port change is removed at `5e5486fadddf819824bdce5d6c8f47383a1e1c79`.

The current PR diff no longer changes `src/lib/onboard/dashboard-port.ts` or its test. PR #9569 remains the single owner for issue #9568, while this pull request remains limited to the OpenShell `0.0.106` migration.

This resolves only the LOC reduction and codebase-simplicity finding. It is not an approval or a judgment on other review dimensions.

**Comment by prekshivyas** [comment]
Status update on `5445d235e8c506db6897eb0c7af6529e71e8fbd1` / base `c1eba30d80916c9afae35310c98c0e81d87ac967`:

- Merged current `main`, including #9579 (the inherited Hermes fixture correction).
- `npm run validate:pr` passes.
- Focused runtime-identity verification passes 135/135 (134 plugin tests plus the lifecycle integration test).
- The latest trusted exact-head inference-routing run ([32221194075](https://github.com/NVIDIA/NemoClaw/actions/runs/32221194075)) did not reach the identity assertions: TC-INF-12 and TC-INF-13 both failed during sandbox onboarding on the dashboard-port race fixed by #9569. This is therefore blocked evidence, not a runtime-identity assertion failure.
- Refreshed product CI is running with no product failures so far; `installer-integration` is green. One advisor lane failed in advisor execution.

Next step: merge #9569, sync `main` again, and dispatch a new trusted exact-head inference-routing E2E. I am intentionally not duplicating #9569 in this PR because that concern was separated during review.

**Review by senthilr-nv — CHANGES_REQUESTED** [review]
Blocking: OpenShell 0.0.106 requires every static credential exposed as an environment placeholder to be classified by endpoint-binding metadata. This PR now requires 0.0.106, but src/lib/actions/sandbox/mcp-bridge-provider-mutation.ts:51 still creates a profile-less generic provider, and src/lib/actions/sandbox/mcp-bridge-policy-render.ts:104-111 emits the endpoint without credential_binding.provider. In v0.0.106 the gateway therefore supplies the credential with no binding and the supervisor rejects the snapshot as an unclassified credential key, so authenticated MCP bridges lose credential projection. The new docs at docs/deployment/set-up-mcp-bridge.mdx:95-99 also describe the old sandbox-scoped behavior instead of this fail-closed requirement. Import or use an endpointless provider profile, bind the generated endpoint to its concrete provider, update the status and docs, and add a v0.0.106 regression proving resolution only on the intended endpoint.

**Review by apurvvkumaria — APPROVED** [review]
Reviewed commit 24d5f55b959e469019f6ba132e986ef9a55e9b62. I found no critical blocker. The OpenShell 0.0.106 dependency is pinned with immutable installer and supervisor provenance, feature gates and credential-boundary evidence match the new contract, and runtime identity now mints credentials after provider attachment with compensating cleanup on failure. Trusted exact-head E2E run 32320521878 completed successfully, including the runtime-identity scenario that previously blocked this migration.

**Review by senthilr-nv — APPROVED** [review]
Reviewed latest PR commit e3a53c96f4bcc53e4e032526d9a3ae28665698fb. Accepted scope is established by #8893. The OpenShell 0.0.106 MCP path now validates an endpointless nemoclaw-mcp-v1 profile, attaches under an unbound capability policy, applies credential_binding.provider before readiness, rejects legacy generic providers for active use while preserving exact cleanup, and removes bound policy before detach. Owning docs and status guidance match the contract. Security review found no blocker across credential handling, ownership, policy restriction, rollback, or dependency integrity. Focused validation passed: MCP/vLLM 147/147, selector 7/7, package contract 4/4, CLI typecheck/build, repository checks, docs, and normal commit/push hooks. Cross-issue sweep found no adjacent fix or contradiction above the confidence floor.

**Review by senthilr-nv — APPROVED** [review]
Reviewed commit ab648f912e5ef3ba6d3bdd89f1c4fced4440b3c2. Accepted scope is established by #8893. The OpenShell 0.0.106 MCP contract now uses the reviewed provider profile and endpoint credential binding; the follow-up preserves legacy generic-provider cleanup while rejecting rebuild before managed state changes. Focused validation passed 75/75 tests plus CLI type-checking. No unresolved correctness, security, architecture, documentation, or automated-feedback blocker remains.
